ONSdigital/census-rm-pubsub-adapter
# Readiness probe and graceful shutdown on consumer failure

# Motivation and Context
K8s needs a readiness probe to know when the app has successfully started.
The app also needs to handle it gracefully if one of it's consumers dies.

# What has changed
* Add readiness indicator by presence of file
* Add readiness and liveness probe to dev k8s manifest
* Add healthcheck to docker compose
* Gracefully shutdown app if a pubsub receiver errors
* Refactor integration tests 

# How to test?
Deploy the app to your dev k8s. Manually edit the pubsub or rabbit config so it cannot start up properly. You shoud see the app does not show as ready when it cannot start correctly.

# Links
https://trello.com/c/OdlCJqh4/901-pubsub-adapter-readiness-liveness-probes-8
